bitnami/superset/templates/flower/deployment.yaml

```diff
@@ -0,0 +1,189 @@
+      {{- end }}
+      {{- if .Values.flower.priorityClassName }}
+      priorityClassName: {{ .Values.flower.priorityClassName | quote }}
+      {{- end }}
+      {{- if .Values.flower.schedulerName }}
+      schedulerName: {{ .Values.flower.schedulerName | quote }}
+      {{- end }}
+      {{- if .Values.flower.topologySpreadConstraints }}
+      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.flower.topologySpreadConstraints "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.flower.podSecurityContext.enabled }}
+      securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.flower.podSecurityContext "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.flower.terminationGracePeriodSeconds }}
+      terminationGracePeriodSeconds: {{ .Values.flower.terminationGracePeriodSeconds }}
+      {{- end }}
+      initContainers:
+        {{- if .Values.waitForDB.enabled }}
+        {{- include "superset.initContainers.waitForDB" . | nindent 8 }}
+        {{- end }}
+        {{- if .Values.waitForRedis.enabled }}
+        {{- include "superset.initContainers.waitForRedis" . | nindent 8 }}
+        {{- end }}
+        {{- if .Values.flower.initContainers }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.flower.initContainers "context" $) | nindent 8 }}
+        {{- end }}
+      containers:
+        - name: celery-flower
+          image: {{ template "superset.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          {{- if .Values.flower.containerSecurityContext.enabled }}
+          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.flower.containerSecurityContext "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.command "context" $) | nindent 12 }}
+          {{- else if .Values.flower.command }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.flower.command "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else if .Values.flower.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.flower.args "context" $) | nindent 12 }}
+          {{- end }}
+          env:
+            {{- include "superset.configure.common" . | nindent 12 }}
+            {{- include "superset.configure.database" . | nindent 12 }}
+            {{- include "superset.configure.redis" . | nindent 12 }}
+            - name: SUPERSET_ROLE
+              value: "celery-flower"
+            - name: FLOWER_UNAUTHENTICATED_API
+              value: "true"
```

Comment:
Why is this hardcoded?

Reply:
No answer

Reply:
I didn't forget about this one. Main issue with flower auth is that it only support authentication via command line:
```
$ celery flower --basic-auth=user:pswd
```

If we added Flower user/password to the chart, it could be potentially exposed with `ps aux`.

Because of that, I would say the recommended method to add authentication if flower wants to be exposed publicly, should be using an additional reverse-proxy with HTTP basic auth.

Reply:
Nevermind, let me test a couple of alternatives first and will implement a solution that does not expose any credentials.